Petulia/Views/Home/Additional/DropDownPicker.swift

```diff
@@ -0,0 +1,133 @@
+  
+  var body: some View {
+    ZStack {
+      // Static row which shows user's current selection
+      HStack {
+        Text(title)
+        Spacer()
+        Text(options[selection])
+          .foregroundColor(Color.black.opacity(0.6))
+        Image(systemName: "chevron.right")
+          .resizable()
+          .aspectRatio(contentMode: .fit)
+          .frame(width: 10, height: 10)
+      }
+      .font(Font.custom("Avenir Next", size: 16).weight(.medium))
+      .padding(.horizontal, 12)
+      .padding(.vertical, 8)
+      .background(Color.white)
+      .onTapGesture {
+        // show the dropdown options
+        withAnimation(Animation.spring().speed(2)) {
+          showOptions = true
+        }
+      }
+      
+      // Drop down options
+      if showOptions {
+        VStack(alignment: .leading, spacing: 4) {
+          Text(title)
+            .font(Font.custom("Avenir Next", size: 16).weight(.semibold))
+            .foregroundColor(.black)
+          if options.count > 5{
+            ScrollView(.horizontal) {
+              HStack {
+                Spacer()
+                ForEach(options.indices, id: \.self) { i in
+                  if i == selection {
+                    Text(options[i])
+                      .font(.system(size: 12))
+                      .padding(.vertical, 8)
+                      .padding(.horizontal, 12)
+                      .background(Color.gray.opacity(0.2))
+                      .cornerRadius(4)
+                      .onTapGesture {
+                        // hide dropdown options - user selection didn't change
+                        withAnimation(Animation.spring().speed(2)) {
+                          showOptions = false
+                        }
+                      }
+                  } else {
+                    Text(options[i])
+                      .font(.system(size: 12))
+                      .onTapGesture {
+                        // update user selection and close options dropdown
+                        withAnimation(Animation.spring().speed(2)) {
+                          selection = i
+                          showOptions = false
+                        }
+                      }
+                  }
+                  Spacer()
+                }
+              }
+              .padding(.vertical, 2)
+              .transition(AnyTransition.move(edge: .top).combined(with: .opacity))
+            }
+          }else{
+            HStack {
+              Spacer()
+              ForEach(options.indices, id: \.self) { i in
+                if i == selection {
+                  Text(options[i])
+                    .font(.system(size: 12))
+                    .padding(.vertical, 8)
+                    .padding(.horizontal, 12)
+                    .background(Color.gray.opacity(0.2))
+                    .cornerRadius(4)
+                    .onTapGesture {
+                      // hide dropdown options - user selection didn't change
```

Comment:
Are the comments part of the code used from the reference? if yes, make sure they match your team's commenting style, if not it's ok I just hadn't seen that many comments